### PR TITLE
Improve CI configuration for better performances

### PR DESCRIPTION
**Pull request by tgalopin**
See https://github.com/puli/issues/issues/169.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -3,10 +3,12 @@ platform: x86
 clone_folder: c:\projects\fos-message
 
 cache:
-  - '%LOCALAPPDATA%\Composer\files'
```
Comment (stof) [inline appveyor.yml:6]: you should keep caching the composer downloads

File: .travis.yml [hunk 2]
```diff
@@ -4,18 +4,25 @@ sudo: false
...
     - php: 5.5
     - php: 5.6
-    - php: 5.6
+    - php: 7.0
+    - php: 7.0
       env: COMPOSER_FLAGS='--prefer-lowest --prefer-stable'
-    - php: 7
     - php: hhvm
+  allow_failures:
+    - php: nightly
```
Comment (stof) [inline .travis.yml]: this is useless as there is no such version in the build

File: appveyor.yml (hunk 1 above)
Comment (tgalopin) [inline appveyor.yml:6]: I'm not sure it's worth it according to nicolas: https://github.com/puli/issues/issues/169#issuecomment-169315809. What's your opinion on this?

File: .travis.yml (hunk 2 above)
Comment (tgalopin) [inline .travis.yml]: Indeed, thanks :) .

**Pull request merged**